ql/src/java/org/apache/hadoop/hive/ql/history/HiveScribeImpl.java

```diff
@@ -0,0 +1,348 @@
+package org.apache.hadoop.hive.ql.history;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.hadoop.hive.ql.QueryPlan;
+import org.apache.hadoop.hive.ql.QueryStats;
+import org.apache.hadoop.hive.ql.exec.Task;
+import org.apache.hadoop.hive.ql.session.SessionState;
+import org.apache.hadoop.mapred.Counters;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HiveScribeImpl implements HiveHistory {
+  private static final Logger LOG = LoggerFactory.getLogger("hive.ql.exec.HiveScribeImpl");
+
+  private Map<String, String> idToTableMap = null;
+
+  // Job Hash Map
+  private final HashMap<String, QueryInfo> queryInfoMap = new HashMap<>();
+
+  // Task Hash Map
+  private final HashMap<String, TaskInfo> taskInfoMap = new HashMap<>();
+
+  private static final String ROW_COUNT_PATTERN = "RECORDS_OUT_(\\d+)(_)*(\\S+)*";
+
+  private static final Pattern rowCountPattern = Pattern.compile(ROW_COUNT_PATTERN);
+
+  // Query Hash Map
+  private HashMap<String, QueryStats> queryStatsMap = new HashMap<>();
+
+  /**
+   * Construct HiveScribeImpl object and scribe query metric to log pipeline.
+   */
+  public HiveScribeImpl(SessionState sessionState) {
+    LOG.info("Instantiated an instance for HiveScribeImpl to scribe query logs.");
+  }
+
+  /**
+   * HistFileName is not used in this impl.
+   */
+  @Override
+  public String getHistFileName() {
+    return "HiveQueryCompletionScribe";
+  }
+
+  /**
+   * Call TwitterScriber to scribe hive query statistics to log pipeline
+   */
+  private void scribeQueryMetricEntry(String queryID) {
+    QueryStats stats = queryStatsMap.get(queryID);
+    if (stats == null) {
+      LOG.info("Stats is null. Nothing passed to log pipeline");
+      return;
+    }
+    QueryCompletedEventScriber hiveHistScriber = new QueryCompletedEventScriber();
```

Comment:
This seems repeatedly creating a new scriber for every query?

Reply:
Changed to `private static`